### PR TITLE
Move all config to 'git-elegant-configure'

### LINKED ISSUE #39: git-elegant:121-121: Move `user.name` and `user.email`...
The puzzle `9-5878affc` (from #9) in [`src/main/git-elegant`](https://github.com/extsoft/elegant-git/blob/master/src/main/git-elegant) (lines 121-121) has to be resolved: "Move `user.name` and `user.email` configuration to the `git-elegant-configure` executable."

The puzzle was created by Dmytro Serdiuk on 23-Aug-17. 

Estimate: 0 minutes, role: IMP.

If you have any technical questions, don't ask me, submit new tickets instead. The task will be "done" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by AntonKarabaza**
Move configuration of `user.name` and `user.email` to separate executable `git-elegant-configure` and remove redundant method `_rewrite_config()`.
Update unit tests for `git-elegant-configure`.
Update commands used `_rewrite_config()`.
Update tests for `init` and `clone` commands.

Closes #39

### CODE REVIEW

**Comment by AntonKarabaza** [comment]
@ivan-podorozhnyi In 870144dac65c10c908aaf79290b4c0ab5f5d7c49 I made the changes that I mentioned. Method `_rewrite_config()` was responsible only for setting two configurations `user.name` and `user.email`. So I moved their configuration logic to `git-elegant-configure` and removed `_rewrite_config()`. But you remind me about `__config` array in `git-elegant` that I had to remove. I'll fix it now.

**Comment by ivan-podorozhnyi** [comment]
@AntonKarabaza good!

File: src/test/git-elegant-configure.bats [hunk 1]
```diff
@@ -6,6 +6,14 @@ load fake-read
 setup() {
     fake-pass git "config --global core.commentChar" "|"
     fake-pass git "config --local core.commentChar" "|"
+
+    fake-pass git "config --global user.name" "UserName"
+    fake-pass git "config --global user.email" "UserEmail"
+
+    fake-pass git "config --global user.name UserName"
```
Comment (extsoft) [inline src/test/git-elegant-configure.bats]: This line is a duplicate.

File: src/test/git-elegant-configure.bats [hunk 2]
```diff
@@ -6,6 +6,14 @@ load fake-read
...
+    fake-pass git "config --global user.email UserEmail"
```
Comment (extsoft) [inline src/test/git-elegant-configure.bats]: This line is a duplicate.

File: src/test/git-elegant-configure.bats (hunk 1 above)
Comment (AntonKarabaza) [reply src/test/git-elegant-configure.bats]: @extsoft 
`fake-pass git "config --global user.name" "UserName"` is used to have default value of user.name. (To mock this `_user_name_default=$(git config --global "$_user_name_key")`

`fake-pass git "config --global user.email" "UserEmail"` is used to have default value of user.email. (To mock this `_user_email_default=$(git config --global "$_user_email_key")`

The `fake-pass git "config --global user.name UserName"` and `fake-pass git "config --global user.email UserEmail"` mock the final git command, this: `git config $MODE $(eval "echo -n \$${f}_key") $answer`

So they are not duplicates. It's a mocks for different commands.

Comment (extsoft) [reply src/test/git-elegant-configure.bats]: @AntonKarabaza oh, okay.

**Comment by extsoft** [comment]
@rultor help

**Comment by extsoft** [comment]
@rultor please merge.

**Pull request merged**